sdk/python/kfp/v2/dsl/type_utils.py

```
@@ -0,0 +1,87 @@
+# The keys are normalized (lowercased). These are types viewed as Parameters.
+# The values are the corresponding IR parameter primitive types.
+_PARAMETER_TYPES_MAPPING = {
+    'integer': pipeline_spec_pb2.PrimitiveType.INT,
+    'int': pipeline_spec_pb2.PrimitiveType.INT,
+    'double': pipeline_spec_pb2.PrimitiveType.DOUBLE,
+    'float': pipeline_spec_pb2.PrimitiveType.DOUBLE,
+    'string': pipeline_spec_pb2.PrimitiveType.STRING,
+    'str': pipeline_spec_pb2.PrimitiveType.STRING,
+}
+
+
+def is_artifact_type(type_name: str) -> bool:
+  """Check if a ComponentSpec I/O type is considered as an artifact type.
+
+  Args:
+    type_name: type name of the ComponentSpec I/O type.
+
+  Returns:
+    True if the type name maps to an artifact type else False.
+  """
+  return type_name.lower() in _ARTIFACT_TYPES_MAPPING
+
+
+def is_parameter_type(type_name: str) -> bool:
+  """Check if a ComponentSpec I/O type is considered as a parameter type.
+
+  Args:
+    type_name: type name of the ComponentSpec I/O type.
+
+  Returns:
+    True if the type name maps to a parameter type else False.
+  """
+  return type_name.lower() in _PARAMETER_TYPES_MAPPING
+
+
+def get_artifact_type(type_name: str) -> str:
+  """Get the IR I/O artifact type for the given ComponentSpec I/O type.
+
+  Args:
+    type_name: type name of the ComponentSpec I/O type.
+
+  Returns:
+     The string value of artifact type if a mapping exists else an empty string.
+  """
+  return _ARTIFACT_TYPES_MAPPING.get(
+      type_name.lower()) if is_artifact_type(type_name) else ''
+
+
+def get_parameter_type(type_name: str) -> pipeline_spec_pb2.PrimitiveType:
+  """Get the IR I/O parameter type for the given ComponentSpec I/O type.
+
+  Args:
+    type_name: type name of the ComponentSpec I/O type.
+
+  Returns:
+    The enum value of the mapped IR I/O primitive type.
+  """
+  return _PARAMETER_TYPES_MAPPING.get(type_name.lower()) if is_parameter_type(
```

Should we actually raise an Error if cannot find a legal mapped type, instead of returning TYPE_UNSPECIFIED?